kngsild/src/main/kotlin/io/egm/kngsild/utils/AuthUtils.kt

```diff
@@ -15,51 +13,60 @@ import java.net.http.HttpRequest
 import java.net.http.HttpResponse
 
 class AuthUtils(
-    private val serverUrl: String,
-    private val clientId: String,
-    private val clientSecret: String,
-    private val grantType: String,
-    private val enabled: Boolean = true
+    private val clientCredentials: ClientCredentials? = null,
+    private val providedToken: ProvidedToken? = null,
+    private val authType: AuthType = AuthType.NONE
 ) {
 
     private val logger = LoggerFactory.getLogger(javaClass)
 
     fun getToken(): Either<ApplicationError, String> {
-        if (!enabled) {
-            logger.debug("Authentication is not enabled, returning random string")
-            return "Unused-Thing".right()
-        }
-
-        val request = HttpRequest
-            .newBuilder()
-            .uri(URI.create(serverUrl))
-            .POST(
-                buildFormDataFromMap(
-                    mapOf(
-                        "client_id" to clientId,
-                        "client_secret" to clientSecret,
-                        "grant_type" to grantType
-                    )
-                )
-            )
-            .header("Content-Type", "application/x-www-form-urlencoded")
-            .build()
+        when (authType) {
+            AuthType.NONE -> {
+                logger.debug("Authentication is not enabled, returning random string")
+                return "Unused-Thing".right()
+            }
+            AuthType.PROVIDED_TOKEN -> {
+                return providedToken?.accessToken?.right()
+                    ?: AuthenticationServerError("ProvidedToken are not set").left()
```

Comment:
you could do this check at construction time (and same for `CLIENT_CREDENTIALS`) and return a configuration error (this is not an error from the authentication server)